### PR TITLE
LinkedHashMap.keySet() order #1850

### PR DESCRIPTION
**Pull request by ruslansennov**

### CODE REVIEW

File: javaslang/src/test/java/javaslang/collection/LinkedHashMapTest.java [hunk 1]
```diff
@@ -164,6 +164,14 @@ public void shouldWrapMap() {
         assertThat(LinkedHashMap.ofAll(source)).isEqualTo(emptyIntInt().put(1, 2).put(3, 4));
     }
 
+    // -- keySet
+
+    @Test
+    public void shouldKeepKeySetOrder() {
+        final Set<Integer> keySet = LinkedHashMap.of(4, "d", 1, "a", 2, "b").keySet();
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/collection/LinkedHashMapTest.java]: :+1:

File: javaslang/src/main/java/javaslang/collection/LinkedHashSet.java [hunk 2]
```diff
@@ -32,9 +32,9 @@
 
     private static final LinkedHashSet<?> EMPTY = new LinkedHashSet<>(LinkedHashMap.empty());
 
-    private final LinkedHashMap<T, T> map;
+    private final LinkedHashMap<T, Object> map;
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/LinkedHashSet.java]: would `<T, ?>` also work, or just `<T, Object>`?

Comment (ruslansennov) [inline javaslang/src/main/java/javaslang/collection/LinkedHashSet.java]: No, '?' fails at compilation stage

Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/LinkedHashSet.java]: k, thanks

**Comment by danieldietrich** [comment]
@ruslansennov looks great, thank you!

**Pull request merged**